logicalplan/propagate_selectors.go

```diff
@@ -21,8 +21,8 @@ func (m PropagateMatchersOptimizer) Optimize(expr parser.Expr) parser.Expr {
 			return
 		}
 
-		// The optimizer cannot be applied to comparison operations.
-		if binOp.Op.IsComparisonOperator() {
+		// The optimizer cannot be applied to comparison operations or 'atan2'.
+		if binOp.Op.IsComparisonOperator() || binOp.Op.String() == "atan2" {
```

Comment:
Interesting, what about other trig functions?

Reply:
I think atan2 is the only one that can be used as a binary expression!